### DISCUSSION TITLE
How to fetch Project UUID just with the project name and version

### DISCUSSION THREAD

**Discussion by aejazuddinam** [DependencyTrack/dependency-track#1207] [Q&A] [answered]
Hi Steve,
Good day, my requirement is for automation, as we know UUID of project is created only after a new project is created.

I have followed this approach for **Getting UUID of project** by using /api/v1/project call and pulling all the projects and parsing them with array & comparing the project name, version created by my jenkins job to fetch its UUID

But this will get much slower when the database gets populated and we have possibly thousands of projects in it.
Do we have a more targeted REST requests with examples rather than pulling back all projects and having to parse them? 

Regards,
Aejaz

**Comment by nscuro** [comment, marked as answer]
Hi @aejazuddinam, the `/api/v1/project/lookup` resource does exactly what you want:

```
curl -H "X-Api-Key: ..." -H "Accept: application/json" \
    'https://dtrack.example.com/api/v1/project/lookup?name=acme-app&version=1.0.0'
```

**Reply by aejazuddinam** [reply]
Thanks @nscuro , this helped.
